ffi/framebuffer_android.lua

```diff
@@ -28,19 +28,11 @@ end
 
 -- update the entire screen
 function framebuffer:_updateFull()
-    -- freescale ntx platform
-    if has_eink_screen and (eink_platform == "freescale" or eink_platform == "qualcomm") then
-        if has_eink_full_support then
-            -- we handle the screen entirely
-            self:_updatePartial(full, delay_page)
-        else
-            -- we're racing against system driver. Let the system win and apply
-            -- a full update after it.
-            self:_updatePartial(full, 500)
-        end
-    -- rockchip rk3x platform
-    elseif has_eink_screen and (eink_platform == "rockchip") then
+    if not has_eink_screen then return end
```

Comment:
It always calls updateWindow or whatever it's called for the ANativeWindowBuffer thing?

Reply:
yes, see https://github.com/koreader/koreader-base/blob/master/ffi/framebuffer_android.lua#L158

Reply:
all partial functions have check for has_eink_full_support.

Wouldn't be better use the same pattern for refreshFullImp?

function framebuffer:refreshFullImp(x, y, w, h) -- luacheck: ignore
    self:_updateWindow()
    if has_eink_screen then
        self:_updateFull()
end

Reply:
yes!